server/src/main/java/org/opensearch/extensions/ExtensionsManager.java

```diff
@@ -407,27 +379,35 @@ public String executor() {
                 return ThreadPool.Names.GENERIC;
             }
         };
-        try {
-            logger.info("Sending extension request type: " + REQUEST_EXTENSION_ACTION_NAME);
-            transportService.connectToExtensionNode(extension);
-            transportService.sendRequest(
-                extension,
-                REQUEST_EXTENSION_ACTION_NAME,
-                new InitializeExtensionRequest(transportService.getLocalNode(), extension),
-                initializeExtensionResponseHandler
-            );
-            inProgressFuture.orTimeout(EXTENSION_REQUEST_WAIT_TIMEOUT, TimeUnit.SECONDS).join();
-        } catch (CompletionException | ConnectTransportException e) {
-            if (e.getCause() instanceof TimeoutException || e instanceof ConnectTransportException) {
-                logger.info("No response from extension to request.", e);
-            } else if (e.getCause() instanceof RuntimeException) {
-                throw (RuntimeException) e.getCause();
-            } else if (e.getCause() instanceof Error) {
-                throw (Error) e.getCause();
-            } else {
-                throw new RuntimeException(e.getCause());
+
+        logger.info("Sending extension request type: " + REQUEST_EXTENSION_ACTION_NAME);
+        transportService.getThreadPool().generic().execute(new AbstractRunnable() {
+            @Override
+            public void onFailure(Exception e) {
+                if (e.getCause() instanceof ConnectTransportException) {
+                    logger.info("No response from extension to request.", e);
+                    throw (ConnectTransportException) e.getCause();
+                } else if (e.getCause() instanceof RuntimeException) {
+                    throw (RuntimeException) e.getCause();
+                } else if (e.getCause() instanceof Error) {
+                    throw (Error) e.getCause();
+                } else {
+                    throw new RuntimeException(e.getCause());
+                }
             }
-        }
+
+            @Override
+            protected void doRun() throws Exception {
+                transportService.connectToExtensionNode(extension);
+                transportService.sendRequest(
+                    extension,
+                    REQUEST_EXTENSION_ACTION_NAME,
+                    new InitializeExtensionRequest(transportService.getLocalNode(), extension),
+                    initializeExtensionResponseHandler
+                );
+                inProgressFuture.orTimeout(EXTENSION_REQUEST_WAIT_TIMEOUT, TimeUnit.SECONDS).join();
+            }
+        });
```

Comment:
The `execute()` terminates and returns here regardless of the status of the other thread.  Suggest using `submit()` instead of `execute()` and you'll get a `Future` you can check whether it's completed.

Reply:
Changed the channel to return a `202` with a message `A request to initialize an extension has been sent.`. Since the extension initialization is done on another transport thread, it's not possible to catch the error either using `execute` or `submit`.